server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -0,0 +1,436 @@
+        clusterSettings.addSettingsUpdateConsumer(
+            CLUSTER_ROUTING_ALLOCATION_AWARENESS_FORCE_GROUP_SETTING,
+            this::setForcedAwarenessAttributes
+        );
+    }
+
+    private void setAwarenessAttributes(List<String> awarenessAttributes) {
+        this.awarenessAttributes = awarenessAttributes;
+    }
+
+    private void setForcedAwarenessAttributes(Settings forceSettings) {
+        Map<String, List<String>> forcedAwarenessAttributes = new HashMap<>();
+        Map<String, Settings> forceGroups = forceSettings.getAsGroups();
+        for (Map.Entry<String, Settings> entry : forceGroups.entrySet()) {
+            List<String> aValues = entry.getValue().getAsList("values");
+            if (aValues.size() > 0) {
+                forcedAwarenessAttributes.put(entry.getKey(), aValues);
+            }
+        }
+        this.forcedAwarenessAttributes = forcedAwarenessAttributes;
+    }
+
+    public void initiateAttributeDecommissioning(
+        final DecommissionAttribute decommissionAttribute,
+        final ActionListener<ClusterStateUpdateResponse> listener,
+        ClusterState state
+    ) {
+        // validates if the correct awareness attributes and forced awareness attribute set to the cluster before initiating decommission
+        // action
+        validateAwarenessAttribute(decommissionAttribute, awarenessAttributes, forcedAwarenessAttributes);
+
+        Set<DiscoveryNode> clusterManagerNodesToBeDecommissioned = nodesWithDecommissionAttribute(state, decommissionAttribute, true);
+        ensureNoQuorumLossDueToDecommissioning(
+            decommissionAttribute,
+            clusterManagerNodesToBeDecommissioned,
+            state.getLastAcceptedConfiguration()
+        );
+
+        DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().custom(DecommissionAttributeMetadata.TYPE);
+        // validates that there's no inflight decommissioning or already executed decommission in place
+        ensureNoAwarenessAttributeDecommissioned(decommissionAttributeMetadata, decommissionAttribute);
+
+        logger.info("initiating awareness attribute [{}] decommissioning", decommissionAttribute.toString());
+
+        // remove all 'to-be-decommissioned' cluster manager eligible nodes from voting config
+        // The method ensures that we don't exclude same nodes multiple times
+        excludeDecommissionedClusterManagerNodesFromVotingConfig(clusterManagerNodesToBeDecommissioned);
+
+        // explicitly throwing NotClusterManagerException as we can certainly say the local cluster manager node will
+        // be abdicated and soon will no longer be cluster manager.
+        if (transportService.getLocalNode().isClusterManagerNode()
+            && !nodeHasDecommissionedAttribute(transportService.getLocalNode(), decommissionAttribute)) {
+            registerDecommissionAttribute(decommissionAttribute, listener);
```

Comment:
As a first step, we are just abdicating a leader which will soon be decommissioned and not actually starting the decommission process. Once we successfully abdicate the to-be-decommissioned leader, we start with metadata. If the abdication itself fails then there's no point of having any metadata around decommissioning. 

Also, as @psychbot pointed out, I'll ensure if INIT fails, we clear off the exclusion

Reply:
Agree I had similar concerns in the [comment](https://github.com/opensearch-project/OpenSearch/pull/4084/files#r963648213)

Reply:
I think any failure/unexpected state should either try to gracefully attempt to take the system to a last known stable state or let it stay in a partially failed state. I would prefer the latter since in an unexpected failure case a roll-back might not be fully possible. The administrator can either attempt a rollback or exercise a kill switch to resurrect the cluster to a normally operating state

Reply:
@shwetathareja @Bukhtawar Thanks for pointing out. Updated in the latest commit that we first register and initiate the metadata and then proceed with exclusion.

> Then, leader abdication could be the first step and the decommissioning process can fail to initialize if abdication fails

We are not reliant on abdication for decommissioning. Here, we just need to ensure that decommissioning is always executed on a master which is never going to be decommissioned and to-be-decommissioned leader are no more part of voting config. Decommissioning fails if the exclusion fails as the cluster manager request will be timed out Updated it to below - 
```
if (transportService.getLocalNode().isClusterManagerNode()
            && !nodeHasDecommissionedAttribute(transportService.getLocalNode(), decommissionAttribute)
            && toBeDecommissionedClusterManagerNodesExcluded
        )
```